xml/favicons.xml

```diff
@@ -95,35 +96,86 @@
+    <param pos="0" name="service.vendor" value="Drupal"/>
+    <param pos="0" name="service.product" value="CMS"/>
+    <param pos="0" name="service.version" value="7.x"/>
+    <param pos="0" name="service.certainty" value="0.5"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:drupal:drupal:7.x"/>
+  </fingerprint>
+
+  <fingerprint pattern="^cf2445dcb53a031c02f9b57e2199bc03$">
+    <description>Drupal CMS (8.0.0)</description>
     <!--
       High definition Drupal favicon
       Drupal 8.0.0 https://github.com/drupal/drupal/blob/2ace26881d7a67a396caea6a79437c7e4e629f98/core/misc/favicon.ico
       Drupal Bartik 9.2+ https://github.com/drupal/drupal/blob/cfa3a480fca997eaecdca2c81c8035ec61308abe/core/themes/bartik/favicon.ico
     -->
-
     <example>cf2445dcb53a031c02f9b57e2199bc03</example>
-    <!-- Drupal 9.2+ https://github.com/drupal/drupal/blob/cfa3a480fca997eaecdca2c81c8035ec61308abe/core/misc/favicon.ico -->
+    <param pos="0" name="service.vendor" value="Drupal"/>
+    <param pos="0" name="service.product" value="CMS"/>
+    <param pos="0" name="service.version" value="8.0.0"/>
+    <param pos="0" name="service.certainty" value="0.5"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:drupal:drupal:8.0.0"/>
+  </fingerprint>
 
+  <fingerprint pattern="^fe22dd2bb09daccf58256611ac491469$">
+    <description>Drupal CMS (9.2+)</description>
+    <!-- Drupal 9.2+ https://github.com/drupal/drupal/blob/cfa3a480fca997eaecdca2c81c8035ec61308abe/core/misc/favicon.ico -->
     <example>fe22dd2bb09daccf58256611ac491469</example>
     <param pos="0" name="service.vendor" value="Drupal"/>
     <param pos="0" name="service.product" value="CMS"/>
+    <param pos="0" name="service.version" value="9.2+"/>
     <param pos="0" name="service.certainty" value="0.5"/>
-    <param pos="0" name="service.cpe23" value="cpe:/a:drupal:drupal:-"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:drupal:drupal:9.2"/>
+  </fingerprint>
+
+  <fingerprint pattern="^e6a9dc66179d8c9f34288b16a02f987e$">
+    <description>Drupal CMS (5.10)</description>
+    <example>e6a9dc66179d8c9f34288b16a02f987e</example>
+    <param pos="0" name="service.vendor" value="Drupal"/>
+    <param pos="0" name="service.product" value="CMS"/>
+    <param pos="0" name="service.version" value="5.10"/>
+    <param pos="0" name="service.certainty" value="0.5"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:drupal:drupal:5.10"/>
+  </fingerprint>
+
+  <fingerprint pattern="^a4819787db1dabe1a6b669d5d6df3bfd$">
+    <description>Drupal CMS (2.x-4.x)</description>
+    <example>a4819787db1dabe1a6b669d5d6df3bfd</example>
+    <param pos="0" name="service.vendor" value="Drupal"/>
+    <param pos="0" name="service.product" value="CMS"/>
+    <param pos="0" name="service.version" value="2.x-4.x"/>
+    <param pos="0" name="service.certainty" value="0.5"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:drupal:drupal:4.x"/>
```

Comment:
I wasn't sure how to best handle the CPE range here so I just used the latest version in the range. Open to any suggestions of course to better interpret this.